### PR TITLE
feat: add internationalization

### PR DESCRIPTION
**Pull request by maltejur**
TODO before merge:

- [x] update all strings to use translations
- [x] add language selector
- [x] (use explanations from other languages)

### CODE REVIEW

File: public/locales/de/index.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
...
+  "aboutUsTitle": "Über uns",
+  "aboutUs": "Wir sind eine Gruppe von Programmierern, die sich gegenseitig helfen, neue Dinge zu bauen, sei es das Schreiben komplexer Verschlüsselungsprogramme oder einfacher Chiffren. Unser Ziel ist es, zusammenzuarbeiten, um schöne, hilfreiche und interessante Algorithmen mit Code zu dokumentieren und zu modellieren. Wir sind eine Open-Source-Gemeinschaft - jeder kann etwas beitragen. Wir überprüfen die Arbeit der anderen, kommunizieren und arbeiten zusammen, um Probleme zu lösen. Wir bemühen uns, einladend und respektvoll zu sein, und stellen dennoch sicher, dass unser Code den neuesten Programmierrichtlinien folgt."
+}
```
Comment (Panquesito7) [inline public/locales/de/index.json]:
```suggestion
}

```

File: README.md [hunk 2]
```diff
@@ -25,10 +25,22 @@
...
 
 This is a static [Next.js](https://nextjs.org/) site providing a searchable library of all the algorithms in [The Algorithms](https://github.com/TheAlgorithms). All the data about the algorithms get fetched from the `DIRECTORY.md` files in all the different repositories.
 
+## Translating the website
+
+You can help us translate the [TheAlgorithms](https://the-algorithms.com) website using [Weblate](https://hosted.weblate.org/engage/TheAlgorithms/?utm_source=widget). If you're not sure about if your translation is OK, better submit a suggestion, and someone else will review it. Here is the list of the current languages:
+
+[![Global translation status](https://hosted.weblate.org/widgets/TheAlgorithms/-/index/287x66-white.png)](https://hosted.weblate.org/engage/TheAlgorithms/?utm_source=widget)
+
+[![Translation status by language](https://hosted.weblate.org/widgets/TheAlgorithms/-/index/multi-auto.svg)](https://hosted.weblate.org/projects/TheAlgorithms/common/?utm_source=widget)
```
Comment (Panquesito7) [inline README.md]:
```suggestion
[![Translation status by language](https://hosted.weblate.org/widgets/TheAlgorithms/-/common/multi-auto.svg)](https://hosted.weblate.org/projects/TheAlgorithms/common/?utm_source=widget)
```

**Comment by Panquesito7** [comment]
> * [ ] (use explanations from other languages)

The Spanish translation for **all** English-based algorithms is done. https://github.com/TheAlgorithms/Algorithms-Explanation/pull/93

File: yarn.lock [hunk 3]
```diff
@@ -640,10 +682,10 @@ ansi-styles@^4.0.0, ansi-styles@^4.1.0:
   dependencies:
     color-convert "^2.0.1"
 
-ansi_up@^5.0.0:
-  version "5.0.0"
-  resolved "https://registry.yarnpkg.com/ansi_up/-/ansi_up-5.0.0.tgz#9bc3b087a59bd543c98dd235a9c8e0a842c454e9"
-  integrity sha512-RHw/w3Kb2U3k4XKfl8FXZW9ldxtTBbLNdKO0RboYeU4ReVwRP77M7b/OxiavMGZsBWcDxn/T0QiR+VtLf7mPYw==
+ansi_up@^4.0.4:
```
Comment (Panquesito7) [inline yarn.lock:685]: `ansi_up` should be at least version 5.0.0, as in the `4.0.4` version there's a medium-severity vulnerability.

Comment (maltejur) [reply yarn.lock:685]: > \>yarn audit
yarn audit v1.22.10
0 vulnerabilities found - Packages audited: 786

Yarn tells me everything is fine. Where did you find this?

Comment (Panquesito7) [reply yarn.lock:685]: Dependabot reported an error previously. Let's see if that happens when we merge this PR...

Comment (maltejur) [reply yarn.lock:685]: okay

**Review by Panquesito7 — APPROVED** [review]
Amazing work; thank you! This works perfectly. Can be merged. 👍 🎉

**Pull request merged**